src/postgres/commands/PostgresAccountWizard/PostgresServerNameStep.ts

```diff
@@ -43,11 +42,18 @@ async function validatePostgresServerName(name: string, client: PostgreSQLManage
     }
 
     const availabilityRequest: NameAvailabilityRequest = { name: name, type: "Microsoft.DBforPostgreSQL" };
+    const characterMatch = name.match(/^(?![-])[a-zA-Z0-9-]*(?<![-])$/);
+    const prefixSuffixMatch = name.match(/^[-]*.*[-]*$/);
     const availability: NameAvailability = (await client.checkNameAvailability.execute(availabilityRequest));
 
     if (!availability.nameAvailable) {
-
-        return availability.message;
+        if (availability.reason === 'AlreadyExists') {
+            return localize('serverNameAvailabilityCheck', 'Server name "{0}" is not available.', name);
+        } else if (prefixSuffixMatch) {
```

Comment:
There's still some weirdness here...
<img width="509" alt="Screen Shot 2020-05-07 at 1 09 39 PM" src="https://user-images.githubusercontent.com/22795803/81340057-0c8e4080-9064-11ea-8e35-cf1a406eb6e3.png">

What if we do this instead?
```suggestion
        } else if (name[0] === '-' || name[name.length - 1] === '-') {
```

I like to avoid regex in cases like this, it can get tricky lol

Reply:
I agree a RegExp is probably overkill in this case, but I would suggest this instead:

```suggestion
        } else if (name.startsWith('-') || name.endsWith('-')) {
```